frontend/app/src/components/dialogs/WatcherDialog.vue

```diff
@@ -0,0 +1,380 @@
+    ]
+  };
+
+  changeEditMode(identifier: string) {
+    this.existingWatchersEdit = {
+      ...this.existingWatchersEdit,
+      [identifier]: !this.existingWatchersEdit[identifier]
+    };
+  }
+
+  deleteWatcher(watcher: string) {
+    let watcherPayload: string[] = [];
+    watcherPayload.push(watcher);
+
+    const { commit } = this.$store;
+    this.$api
+      .deleteWatcher(watcherPayload)
+      .then(updatedWatchers => {
+        commit('balances/watchers', updatedWatchers);
+        commit('setMessage', {
+          title: 'Watcher Success',
+          description: 'Successfully deleted the watcher.',
+          success: true
+        } as Message);
+        this.watcherValue = null;
+        this.watcherOperation = null;
+        this.loadedWatchers = JSON.parse(JSON.stringify(updatedWatchers));
+      })
+      .catch((reason: Error) => {
+        commit('setMessage', {
+          title: 'Watcher Error',
+          description: reason.message || 'Error deleting the watcher',
+          success: false
+        } as Message);
+      });
+  }
+
+  editWatcher(watcher: Watcher) {
+    if (this.existingWatchersEdit[watcher.identifier] === false) {
+      // If we're not in edit mode, just go into edit mode
+      this.changeEditMode(watcher.identifier);
+    } else {
+      // If we're in edit mode, check to see if the values have changed before
+      // sending an API call
+      const existingWatcherArgs = this.existingWatchers.find(
+        existingWatcher => existingWatcher.identifier === watcher.identifier
+      )!.args as WatcherArgs['makervault_collateralization_ratio'];
+      const modifiedWatcherArgs = watcher.args as WatcherArgs['makervault_collateralization_ratio'];
+
+      if (
+        existingWatcherArgs.op !== modifiedWatcherArgs.op ||
+        existingWatcherArgs.ratio !== modifiedWatcherArgs.ratio
+      ) {
+        let watcherPayload: Watcher[] = [];
+        watcherPayload.push(watcher);
+
+        const { commit } = this.$store;
+        this.$api
+          .editWatcher(watcherPayload)
+          .then(updatedWatchers => {
+            commit('balances/watchers', updatedWatchers);
+            commit('setMessage', {
+              title: 'Watcher Success',
+              description: 'Successfully edited the watcher.',
+              success: true
+            } as Message);
+            this.changeEditMode(watcher.identifier);
+            this.loadedWatchers = JSON.parse(JSON.stringify(updatedWatchers));
+          })
+          .catch((reason: Error) => {
+            commit('setMessage', {
+              title: 'Watcher Error',
+              description: reason.message || 'Error editing the watcher',
+              success: false
+            } as Message);
+          });
+      } else {
+        this.changeEditMode(watcher.identifier);
+      }
+    }
+  }
+
+  addWatcher() {
+    let watcherPayload: Omit<Watcher, 'identifier'>[] = [];
+
+    if (
```

Comment:
if you don't have an else clause maybe it would make sense to reverse the boolean condition and return, to avoid the unneeded indentation in the following code